spark-cobol/src/main/scala/za/co/absa/cobrix/spark/cobol/utils/SparkUtils.scala

```diff
@@ -70,49 +73,102 @@ object SparkUtils {
       name
     }
 
-    def flattenStructArray(path: String, structField: StructField, arrayType: ArrayType): Unit = {
+    /**
+      * For an array of arrays of arrays, ... get the final element type at the bottom of the array
+      *
+      * @param arrayType An array data type from a Spark dataframe schema
+      * @return A non-array data type at the bottom of array nesting
+      */
+    @tailrec
+    def getDeepestArrayType(arrayType: ArrayType): DataType = {
```

Comment:
Did you decide to keep this in even though you aren't using it?

Reply:
Fair point, will remove that